### ISSUE TITLE
if score is None, score.log_norm_score can't evaluate.

### ISSUE THREAD

**Issue by russelljjarvis** [scidash/sciunit#128]
If not present already, sciunit needs some kind of internal case handling for score is None, when trying to compute the lognorm of the score. This is to help None scores propogate through Sciunit/Neuronunit/ and optimizer code.

In the example code None score protection/testing should possibly be moved into NU internals.
```python
            scores_ = []
            for t in dtc.tests:
                score = t.judge(model)
                if score is not None:
                    score = t.judge(model)
                    score.log_norm_score
                scores_.append(score)
            dtc.SA = ScoreArray(dtc.tests, scores_)
            dtc.SA = dtc.ordered_score()
```
https://github.com/russelljjarvis/NeuronunitOpt/blob/master/neuronunit/optimisation/optimization_management.py#L2208-L2227
Note, a different issue, the code iteration above can be replaced by using the model to slice into sciunit testing.

**Comment by rgerkin** [comment]
@russelljjarvis What values would be acceptable?  I believe it currently gives -Inf, which is the right answer.  There is no way to guarantee an smooth, monotonic, sloped error gradient across all possible values of the score, but if you are willing to accept a flat gradient at very extreme values, then I could just clip it to something like [1e-99, 0] instead of [-Inf, 0].  So let me know if the problem is simply handling infinity, or something deeper.

**Comment by russelljjarvis** [comment]
It's not so much about the gradient, but more just sciunit score throws error when it raw_score is zero and it tries to compute log_norm. I think the intended value is neg infinity. 

I will keep checking perhaps the problem is only on my end.

**Comment by rgerkin** [comment]
It's possible this was a problem earlier and is now fixed, because in this example:
```
import numpy as np
from sciunit.scores import ZScore
score = ZScore(np.inf)
print(score.norm_score, score.log_norm_score)
```
I see that the `norm_score` is 0.0 and the `log_norm_score` is -inf, rather than an error.  It does generate a warning, but that can be ignored.

**Comment by russelljjarvis** [comment]
Bump

**Comment by rgerkin** [comment]
@russelljjarvis 
Are you referring to a `NoneScore`?  In that case, I see that `score.norm_score` was evaluating to `None` and `score.log_norm_score` throws an exception.  I've fixed it in 13697135ae47c76b5e34583a99be21eabfb2ea60 (dev branch) so that `score.log_norm_score` also evaluates to `None` now.  If you need a numeric value after that you should handle it in the optimizer (e.g. by treating `None` as -Inf), but at least there won't be an exception inside of sciunit.
